syntax = "proto3";
import "protobuf/person.proto";

message Location {
    int32 id = 1;
    string person_id = 2;
    string longitude = 3;
    string latitude = 4;
    string creation_time = 5;
}

message CreateLocationRequest {
    string person_id = 1;
    string longitude = 2;
    string latitude = 3;
    string creation_time = 4;
}

message GetLocationRequest {
    int32 id = 1;
}


service LocationService {
    rpc CreateLocation (CreateLocationRequest) returns (Location);
    rpc GetLocation (GetLocationRequest) returns (Location);

}

message Connection {
    api.person.Person person = 1;
    Location location = 2;
}

message ConnectionResponse {
    repeated Connection connections = 1;
}

message GetConnectionRequest {
    int32 person_id = 1;
    string start_date = 2; 
    string end_date = 3;
    float meters = 4;
}

service ConnectionService {
    rpc GetConnection (GetConnectionRequest) returns (ConnectionResponse);
}
